src/main/java/com/algo/nth_element/NthElementUtil.java

```
@@ -0,0 +1,98 @@
+package com.algo.nth_element;
+
+import java.lang.reflect.Array;
+import java.util.Arrays;
+
+public class NthElementUtil {
+    public static <T extends Comparable<T>> void nthElement(T[] array, int n) {
+        if (array == null || n < 0 || n >= array.length) {
```

Кажется, что массив на нуллабилити можно не проверять. Такие проверки нужны только тогда, когда сам объект мы не трогаем, а лишь сохраняем куда-то. В первом случае смысла в проверке нет, так как мы и так получим NPE почти сразу, а во втором случае этот null может далеко улететь от истинного источника проблемы, так что лучше fail fast.